test/views/pages/home/content/ranking/static_ranking_list_test.dart

```diff
@@ -0,0 +1,110 @@
+import "package:flutter/material.dart";
+import "package:flutter_test/flutter_test.dart";
+import "package:proxima/views/components/content/publication_header/user_profile_pop_up.dart";
+import "package:proxima/views/pages/home/content/ranking/ranking_card.dart";
+import "package:proxima/views/pages/home/content/ranking/ranking_list.dart";
+
+import "../../../../../mocks/data/ranking_data.dart";
+
+void main() {
+  testWidgets(
+    "Check that all the ranking elements are displayed correctly",
+    (tester) async {
+      final rankingListWidget = MaterialApp(
+        home: RankingList(
+          rankingDetails: mockRankingDetailsWithtestUser,
+          onRefresh: () async {},
+        ),
+      );
+
+      await tester.pumpWidget(rankingListWidget);
+      await tester.pumpAndSettle();
+
+      // Check that the ranking list is displayed
+      final rankingList = find.byType(RankingList);
+      expect(rankingList, findsOneWidget);
+
+      // Check that the ranking cards are displayed
+      // The additional card is for the current user's ranking card.
+      final rankingCards = find.byType(RankingCard);
+      expect(
+        rankingCards,
+        findsNWidgets(mockRankingElementDetailsList.length),
+      );
+
+      // Check that the user ranks are displayed
+      final userRankTexts = find.byKey(RankingCard.userRankrankTextKey);
+      expect(
+        userRankTexts,
+        findsNWidgets(mockRankingElementDetailsList.length),
+      );
+
+      // Check that the user avatar are displayed
+      final userAvatars = find.byKey(RankingCard.userRankAvatarKey);
+      expect(
+        userAvatars,
+        findsNWidgets(mockRankingElementDetailsList.length),
+      );
+
+      // Check that the display names are displayed
+      final userDisplayNameTexts =
+          find.byKey(RankingCard.userRankDisplayNameTextKey);
+      expect(
+        userDisplayNameTexts,
+        findsNWidgets(mockRankingElementDetailsList.length),
+      );
+
+      // Check that the centauri points are displayed
+      final centauriPointsTexts =
+          find.byKey(RankingCard.userRankCentauriPointsTextKey);
+      expect(
+        centauriPointsTexts,
+        findsNWidgets(mockRankingElementDetailsList.length),
+      );
+
+      //Check that the card color corresponds to the expected color
```

Comment:
We should not test for exact match of colors. This test doesn't provide any security as you are not testing external behavior. You just check that what you did is what you did.

Reply:
If you really want to check for UI colors:

A better test would simply check that the first second third and forth element don't all have the same color (without exact math on the color). And that from forth up to last all have the same color.

That way, you only test external behavior.
A general rule of thumb is that a test should not be modified when we simply change a few visual looks (like a color).

Reply:
I didn't know about this rule of thumb 👍🏼 . Thank you for your comment 💯 

I am going to completely remove the color checking as we may want in the future to modify the color or also add colors to the 4th and 5th places, for instance, or other more complex behavior with colors within the ranking page.

And you are right that it is just a test that verify that the color of the card is the color that I set x)

Removed in: [bcc43d5](https://github.com/ProximaEPFL/proxima/pull/322/commits/bcc43d560ffab941933b3b64ec3210961b11ef0e)

Thank you !